cmd/srcd/cmd/root.go

```diff
@@ -88,37 +88,89 @@ func initConfig() {
 
 var logMsgRegex = regexp.MustCompile(`.*msg="(.+)"`)
 
-func logAfterTimeout(header string) chan struct{} {
-	logs, err := daemon.GetLogs()
-	if err != nil {
-		logrus.Errorf("could not get logs from server container: %v", err)
+func logAfterTimeout(msg string, timeout time.Duration) func() {
+	d := &Defered{
+		Timeout: timeout,
+		Msg:     msg,
 	}
 
-	started := make(chan struct{})
-	go func() {
-		defer logs.Close()
+	return d.Print()
+}
+
+func logAfterTimeoutWithSpinner(msg string, timeout time.Duration, spinnerInterval time.Duration) func() {
+	d := &Defered{
+		Timeout:         timeout,
+		Msg:             msg,
+		Spinner:         true,
+		SpinnerInterval: spinnerInterval,
+	}
+
+	return d.Print()
+}
+
+func logAfterTimeoutWithServerLogs(msg string, timeout time.Duration) func() {
+	logs, err := daemon.GetLogs()
```

Comment:
here 2 problems:
1. you call `GetLogs` even in a case you don't need to
2. you don't close them if inputFn is never called

maybe it makes sense to just move this line into `inputFn`

Reply:
You're right, I moved it inside `inputFn`.